spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -672,129 +623,57 @@ private String toDisplayString(String original, int maxCharacters) {
-
-		private final String topicName;
+	private final static class ProducerConfigurationMessageHandler implements MessageHandler, Lifecycle {
 
-		private final ExtendedProducerProperties<KafkaProducerProperties> producerProperties;
+		private ProducerConfiguration<byte[], byte[]> delegate;
 
-		private final int numberOfKafkaPartitions;
+		private String targetTopic;
 
-		private final ProducerConfiguration<byte[], byte[]> producerConfiguration;
+		private boolean running = true;
 
-		private final PartitionHandler partitionHandler;
+		private ProducerConfigurationMessageHandler(
+				ProducerConfiguration<byte[], byte[]> delegate, String targetTopic) {
+			Assert.notNull(delegate, "Delegate cannot be null");
+			Assert.hasText(targetTopic, "Target topic cannot be null");
+			this.delegate = delegate;
+			this.targetTopic = targetTopic;
 
-		private SendingHandler(String topicName, ExtendedProducerProperties<KafkaProducerProperties> properties,
-				int numberOfPartitions, ProducerConfiguration<byte[], byte[]> producerConfiguration) {
-			this.topicName = topicName;
-			producerProperties = properties;
-			this.numberOfKafkaPartitions = numberOfPartitions;
-			ConfigurableListableBeanFactory beanFactory = KafkaMessageChannelBinder.this.getBeanFactory();
-			this.setBeanFactory(beanFactory);
-			this.producerConfiguration = producerConfiguration;
-			this.partitionHandler = new PartitionHandler(beanFactory, evaluationContext, partitionSelector, properties);
 		}
 
 		@Override
-		protected void handleMessageInternal(Message<?> message) throws Exception {
-			int targetPartition;
-			if (producerProperties.isPartitioned()) {
-				targetPartition = this.partitionHandler.determinePartition(message);
-			}
-			else {
-				targetPartition = roundRobin() % numberOfKafkaPartitions;
-			}
-			if (HeaderMode.embeddedHeaders.equals(producerProperties.getHeaderMode())) {
-				MessageValues transformed = serializePayloadIfNecessary(message);
-				byte[] messageToSend = embeddedHeadersMessageConverter.embedHeaders(transformed,
-						KafkaMessageChannelBinder.this.headersToMap);
-				producerConfiguration.send(topicName, targetPartition, null, messageToSend);
-			}
-			else if (HeaderMode.raw.equals(producerProperties.getHeaderMode())) {
-				Object contentType = message.getHeaders().get(MessageHeaders.CONTENT_TYPE);
-				if (contentType != null && !contentType.equals(MediaType.APPLICATION_OCTET_STREAM_VALUE)) {
-					logger.error("Raw mode supports only " + MediaType.APPLICATION_OCTET_STREAM_VALUE + " content type"
-							+ message.getPayload().getClass());
-				}
-				if (message.getPayload() instanceof byte[]) {
-					producerConfiguration.send(topicName, targetPartition, null, (byte[]) message.getPayload());
-				}
-				else {
-					logger.error("Raw mode supports only byte[] payloads but value sent was of type "
-							+ message.getPayload().getClass());
-				}
-			}
+		public void start() {
```

Comment:
I noticed `running` is initialized to `true` thus this is a no-op... assume that's intentional, but thought I'd point it out since it's slightly non-intuitive